src/arrayMethodSplice.js

```diff
@@ -5,7 +5,56 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const objectLength = this.length;
+    const argLength = items.length;
+    let removedItems = [];
+    let temp = [];
+
+    if (start < 0) start += objectLength;
+    if ((start * -1) > objectLength) {
+      removedItems = [...this];
+      this.length = 0;
+      return removedItems;
+    }
+    if (start === 0) return this;
+
+    if (deleteCount > objectLength) deleteCount = 0;
+    if (deleteCount === 0 && (!items)) return removedItems;
+
+    for (let i = objectLength; i > start; i--) {
+      removedItems.push(this.pop());
+    }
+
+    if (deleteCount && argLength <= 0) {
+      removeFromBuffer();
+      putTogether(this);
+    }
+    if (argLength > 0) {
+      removeFromBuffer();
+      insertArgs(this);
+      putTogether(this);
+    }
+
+    function removeFromBuffer() {
```

Comment:
if you need helper functions create pure functions without side effects outside a main function. Do not declare them inside applyCustomSplice

Reply:
there is no functions inside